File: backups/capacitor-sora-editor-1.3.13/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/lang/util/PlainTextAnalyzeManager.java

package io.github.abc15018045126.sora.lang.util;

import androidx.annotation.NonNull;

import io.github.abc15018045126.sora.lang.styling.Styles;
import io.github.abc15018045126.sora.text.CharPosition;

/**
 * This class generate plain spans for content, in case when
 * using a language without analysis you can still provide context-free completions.
 * By default, editor does not show completions when no span is set
 * on the text. This would be helpful for enabling completion for pure texts.
 *
 * @author abc15018045126
 */
public final class PlainTextAnalyzeManager extends BaseAnalyzeManager {

    @Override
    public void insert(@NonNull CharPosition start, @NonNull CharPosition end, @NonNull CharSequence insertedContent) {

    }

    @Override
    public void delete(@NonNull CharPosition start, @NonNull CharPosition end, @NonNull CharSequence deletedContent) {

    }

    @Override
    public void rerun() {
        final var receiver = getReceiver();
        final var ref = getContentRef();
        if (receiver != null && ref != null) {
            var style = new Styles();
            style.spans = new PlainTextSpans(ref.getLineCount());
            receiver.setStyles(this, style);
        } else if (receiver != null) {
            receiver.setStyles(this, null);
        }
    }

}

File: capacitor-sora-editor/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/lang/util/PlainTextAnalyzeManager.kt
package io.github.abc15018045126.sora.lang.util

import io.github.abc15018045126.sora.lang.styling.Styles
import io.github.abc15018045126.sora.text.CharPosition

/**
 * This class generate plain spans for content, in case when
 * using a language without analysis you can still provide context-free completions.
 * By default, editor does not show completions when no span is set
 * on the text. This would be helpful for enabling completion for pure texts.
 *
 * @author abc15018045126
 */
class PlainTextAnalyzeManager : BaseAnalyzeManager() {

    override fun insert(start: CharPosition, end: CharPosition, insertedContent: CharSequence) {}

    override fun delete(start: CharPosition, end: CharPosition, deletedContent: CharSequence) {}

    override fun rerun() {
        val receiver = receiver
        val ref = contentRef
        if (receiver != null && ref != null) {
            val style = Styles()
            style.spans = PlainTextSpans(ref.lineCount)
            receiver.setStyles(this, style)
        } else receiver?.setStyles(this, null)
    }

}
